src/pykeen/pipeline/api.py

```diff
@@ -567,10 +558,16 @@ def save_pipeline_results_to_directory(
         )
         for epoch, loss in enumerate(pipeline_result.losses):
             losses_rows.append((i, epoch, loss))
+        metrics.append({k: pipeline_result.get_metric(key=k) for k in metrics[0].keys()})
```

Comment:
I don't really understand what's going here... these numeric keys weren't what I was thinking when I said use a dictionary. Like the `metrics` variable itself should be a dictionary, or some other data structure that makes it easier to understand what's inside and index it

Reply:
First, `metrics` is only a local variable, and not passed outside. Instead, it is used to construct the dataframe `metric_df`. This df should look like the following:

| replicate | hits@10 | mr |
| -- | -- | -- |
| original | 0.64  | 123 |
| replicate-1 | 0.63  | 127 |
| replicate-2 | 0.67  | 110 |

i.e. one row per model (original + replicates from reproduction), and one column for each metric which the original paper/configuration reported.

Reply:
Maybe this should also report diffs against original for each metric in the replicates

Reply:
I thought about generating a further summary, but the table above seems to contain all necessary information.

Reply:
true. The other thing I'd ask about is if we could present a p-value for a 1 sample t-test for each metric vs. the "original"